hudi-spark-datasource/hudi-spark/src/test/java/org/apache/hudi/table/upgrade/TestUpgradeDowngrade.java

```diff
@@ -153,6 +156,42 @@ public void testAutoUpgradeDisabled(HoodieTableVersion originalVersion) throws E
     LOG.info("Auto-upgrade disabled test passed for version {}", originalVersion);
   }
 
+  @ParameterizedTest
+  @MethodSource("metadataTableCorruptionTestVersionPairs")
+  public void testMetadataTableUpgradeDowngradeFailure(HoodieTableVersion fromVersion, HoodieTableVersion toVersion) throws Exception {
```

Comment:
@nsivabalan please see the following test, this class is now a new class that has been added from the following PR: https://github.com/apache/hudi/pull/13669

Let me know if its more intuitive, and if there is other cases you want covered.